systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -183,6 +184,27 @@ void testKafkaAndZookeeperScaleUpScaleDown() {
         assertNoCoErrorsLogged(TimeMeasuringSystem.getDurationInSecconds(testClass, testName, operationID));
     }
 
+    @Test
+    @Tag(REGRESSION)
+    void testEODeletion() {
+        // Deploy kafka cluster with EO
+        Kafka kafka = resources().kafkaEphemeral(CLUSTER_NAME, 3).done();
+
+        // Get pod name to check termination process
+        String podName = client.pods().inNamespace(kubeClient.namespace()).list().getItems()
+                .stream().filter(p -> p.getMetadata().getName().startsWith(entityOperatorDeploymentName(CLUSTER_NAME)))
+                .findFirst().get().getMetadata().getName();
```

Comment:
If something goes wrong should we assert that the `findFirst` has returned a non-empty `Optional` so that the EO is really created? Otherwise, it will throw an NPE.

Reply:
Good point, Added check for non-empty `Optional`